### ISSUE TITLE
when I change the difficulty of the map '8m',the reward and win rate of the training process be zero???

### ISSUE THREAD

**Issue by hxx1996** [starry-sky6688/StarCraft#3]
train_hyper-parameter:
map 8m
map_difficulty 4
epoch 20000
replay_buffer 4000
learning_rate 0.0005
epoch: 0 avg_reward: 0.020390625 avg_winrate: 0.0 test_avg_reward 4.3765625 test_avg_winrate 0.0
epoch: 100 avg_reward: 1.8874609375 avg_winrate: 0.0 test_avg_reward 17.509375 test_avg_winrate 0.7
epoch: 200 avg_reward: 1.9575390625 avg_winrate: 0.0 test_avg_reward 15.9609375 test_avg_winrate 0.55
epoch: 300 avg_reward: 2.0929296875 avg_winrate: 0.0 test_avg_reward 17.3375 test_avg_winrate 0.7
epoch: 400 avg_reward: 2.187265625 avg_winrate: 0.0 test_avg_reward 16.646875 test_avg_winrate 0.6
epoch: 500 avg_reward: 2.3000390625 avg_winrate: 0.0 test_avg_reward 15.15625 test_avg_winrate 0.45
epoch: 600 avg_reward: 2.377578125 avg_winrate: 0.0 test_avg_reward 14.1015625 test_avg_winrate 0.3
epoch: 700 avg_reward: 2.59359375 avg_winrate: 0.0 test_avg_reward 16.3703125 test_avg_winrate 0.55
epoch: 800 avg_reward: 2.675390625 avg_winrate: 0.0 test_avg_reward 11.3515625 test_avg_winrate 0.05
epoch: 900 avg_reward: 2.4835546875 avg_winrate: 0.0 test_avg_reward 6.2828125 test_avg_winrate 0.0
epoch: 1000 avg_reward: 2.2533203125 avg_winrate: 0.0 test_avg_reward 4.5921875 test_avg_winrate 0.0
epoch: 1100 avg_reward: 2.1725390625 avg_winrate: 0.0 test_avg_reward 5.3140625 test_avg_winrate 0.0
epoch: 1200 avg_reward: 2.209765625 avg_winrate: 0.0 test_avg_reward 6.115625 test_avg_winrate 0.0
epoch: 1300 avg_reward: 2.2476953125 avg_winrate: 0.0 test_avg_reward 3.9359375 test_avg_winrate 0.0
epoch: 1400 avg_reward: 2.22125 avg_winrate: 0.0 test_avg_reward 3.5078125 test_avg_winrate 0.0
epoch: 1500 avg_reward: 2.1265234375 avg_winrate: 0.0 test_avg_reward 3.2515625 test_avg_winrate 0.0
epoch: 1600 avg_reward: 2.09265625 avg_winrate: 0.0 test_avg_reward 3.365625 test_avg_winrate 0.0
epoch: 1700 avg_reward: 2.0511328125 avg_winrate: 0.0 test_avg_reward 3.478125 test_avg_winrate 0.0
epoch: 1800 avg_reward: 2.0827734375 avg_winrate: 0.0 test_avg_reward 3.1734375 test_avg_winrate 0.0
epoch: 1900 avg_reward: 2.1476171875 avg_winrate: 0.0 test_avg_reward 2.9421875 test_avg_winrate 0.0
epoch: 2000 avg_reward: 2.2252734375 avg_winrate: 0.0 test_avg_reward 3.19375 test_avg_winrate 0.0
epoch: 2100 avg_reward: 2.254609375 avg_winrate: 0.0 test_avg_reward 2.66875 test_avg_winrate 0.0
epoch: 2200 avg_reward: 2.1873828125 avg_winrate: 0.0 test_avg_reward 2.2515625 test_avg_winrate 0.0
epoch: 2300 avg_reward: 2.186484375 avg_winrate: 0.0 test_avg_reward 2.3671875 test_avg_winrate 0.0
epoch: 2400 avg_reward: 2.25515625 avg_winrate: 0.0 test_avg_reward 2.2703125 test_avg_winrate 0.0
epoch: 2500 avg_reward: 2.1551953125 avg_winrate: 0.0 test_avg_reward 1.6890625 test_avg_winrate 0.0
epoch: 2600 avg_reward: 2.086796875 avg_winrate: 0.0 test_avg_reward 1.9234375 test_avg_winrate 0.0
epoch: 2700 avg_reward: 2.277734375 avg_winrate: 0.0 test_avg_reward 4.878125 test_avg_winrate 0.0
epoch: 2800 avg_reward: 2.7228515625 avg_winrate: 0.0 test_avg_reward 4.096875 test_avg_winrate 0.0
epoch: 2900 avg_reward: 2.968671875 avg_winrate: 0.0 test_avg_reward 3.5453125 test_avg_winrate 0.0
epoch: 3000 avg_reward: 3.369453125 avg_winrate: 0.0 test_avg_reward 4.946875 test_avg_winrate 0.0
epoch: 3100 avg_reward: 3.9022265625 avg_winrate: 0.0 test_avg_reward 5.396875 test_avg_winrate 0.0
epoch: 3200 avg_reward: 4.7571875 avg_winrate: 0.0 test_avg_reward 6.01875 test_avg_winrate 0.0

**Comment by starry-sky6688** [comment]
Different maps have different hyper-parameter settings. I find that your test_avg_winrate increases at first, and after some epochs begins to decrease, so I suggest you change the hyper-parameters  about epsilon, which is great important for training.

**Comment by hxx1996** [comment]
> Different maps have different hyper-parameter settings. I find that your test_avg_winrate increases at first, and after some epochs begins to decrease, so I suggest you change the hyper-parameters about epsilon, which is great important for training.

thank you~I will check my code
